src/pages/Test/index.js

```diff
@@ -0,0 +1,154 @@
+import { useRef, useEffect, useState } from "react";
+import Countdown from "react-countdown";
+import { useParams } from "react-router";
+import { toast } from "react-toastify";
+import { getTestDetailApi } from "../../apis/test";
+import { submitTestApi } from "../../apis/testResult";
+import styles from "./Test.module.css";
+
+function Test() {
+  const startDate = useRef(Date.now());
+
+  const { testId } = useParams();
+  const [testDetail, setTestDetail] = useState({
+    _id: testId,
+    content: "Test",
+    timeLimit: 60,
+    questions: []
+  });
+
+  const [results, setResults] = useState({});
+  const [checked, setChecked] = useState({});
+
+  const getTestDetail = () => {
+    getTestDetailApi(testId).then((data) => {
+      if (data.error) {
+        toast.error(data.message);
+      } else {
+        const test = data?.data;
```

Comment:
const test = data?.data;
if there is no data.data, meaning that test = undefined, the rest of code will be error.
 
Also, data.data looks confused, should use **res** or **response** instead data

Reply:
@anhtuyetdut